core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -313,6 +313,9 @@ object NonEmptyList extends NonEmptyListInstances {
       case h :: t => NonEmptyList(h, t)
     }
 
+  def fromFoldable[F[_], A](fa: F[A])(implicit F: Foldable[F]): Eval[Option[NonEmptyList[A]]] =
+    F.reduceRightToOption(fa)(x => NonEmptyList.of(x))((e,acc) => Later(e :: acc.value))
```

Comment:
What about `NonEmptyList.fromList(F.toList(fa))` ?

The `Foldable#toList` implementation using a `mutable.ListBuffer`, which is overridden in most `Foldable` instances (eg `List` just returns the list itself, `Vector` calls `toList`, etc), should be more efficient than prepending a `NonEmptyList`.